src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala

```diff
@@ -26,23 +26,68 @@ object DictFunctions extends RegistryFunctions {
           default)))
   }
 
-  def registerAll() {
-    registerIR("toDict", TArray(tv("T")))(ToDict)
+  val tdict = TDict(tv("key"), tv("value"))
 
-    registerIR("size", TDict(tv("T"), tv("U"))) { d =>
+  def registerAll() {
+    registerIR("size", tdict) { d =>
       ArrayLen(ToArray(d))
     }
 
-    registerIR("isEmpty", TDict(tv("T"), tv("U"))) { d =>
+    registerIR("isEmpty", tdict) { d =>
       ArrayFunctions.isEmpty(ToArray(d))
     }
 
-    registerIR("contains", TDict(tv("K"), tv("V")), tv("K"))(contains)
+    registerIR("contains", tdict, tv("key"))(contains)
 
-    registerIR("get", TDict(tv("K"), tv("V")), tv("K"), tv("V"))(get)
+    registerIR("get", tdict, tv("key"), tv("value"))(get)
 
-    registerIR("get", TDict(tv("K"), tv("V")), tv("K")) { (d, k) =>
+    registerIR("get", tdict, tv("key")) { (d, k) =>
       get(d, k, NA(types.coerce[TDict](d.typ).valueType))
     }
+
+    registerIR("dictToArray", tdict) { d =>
+      val elt = Ref(genUID(), -types.coerce[TContainer](d.typ).elementType)
+      ArrayMap(
+        ToArray(d),
+        elt.name,
+        MakeTuple(Seq(GetField(elt, "key"), GetField(elt, "value"))))
+    }
+
+    registerIR("keySet", tdict) { d =>
+      val pairs = Ref(genUID(), -types.coerce[TContainer](d.typ).elementType)
+      ToSet(ArrayMap(ToArray(d), pairs.name, GetField(pairs, "key")))
+    }
+
+    registerIR("dict", TSet(TTuple(tv("key"), tv("value"))))(s => ToDict(ToArray(s)))
+
+    registerIR("dict", TArray(TTuple(tv("key"), tv("value"))))(ToDict)
+
+    registerIR("Dict", TArray(tv("key")), TArray(tv("value"))) { (k, v) =>
```

Comment:
Can you clarify why this one is capitalized as "Dict" when the other conversions are "dict" ?

Reply:
this is a relic of the old string expr language, I think. no reason to keep it around.

Reply:
you mean this function, or the naming? if the former, I'll remove it from AST.

Reply:
the capitalization, I meant. wasn't clear.

Reply:
is this actually used though?

Reply:
I'll check.

Reply:
I don't have a view either way on whether it should exist.  It just seemed that if the other
conversions are now called lower-case "dict", that should be the name for this one as well.

I guess if the new language doesn't allow it to be called, then it should just disappear.  I
don't know if that's true.

Reply:
ok @rcownie I've removed `Dict` from everything since it's not exposed in Python.